Terminals.cs

```diff
@@ -0,0 +1,525 @@
+          if (string.CompareOrdinal(lexeme, "RETURN") == 0) {
+            return Token.RETURN;
+          } /* end if */
+          break;
+
+        case 'T' :
+          switch (lexeme[0]) {
+
+            case 'E' :
+              /* EXPORT */
+              if (string.CompareOrdinal(lexeme, "EXPORT") == 0) {
+                return Token.EXPORT;
+              } /* end if */
+              break;
+
+            case 'I' :
+              /* IMPORT */
+              if (string.CompareOrdinal(lexeme, "IMPORT") == 0) {
+                return Token.IMPORT;
+              } /* end if */
+              break;
+
+            case 'R' :
+              /* REPEAT */
+              if (string.CompareOrdinal(lexeme, "REPEAT") == 0) {
+                return Token.REPEAT;
+              } /* end if */
+              break;
+
+          } /* end switch */
+          break;
+
+      } /* end switch */
+      break;
+
+    case /* length = 7 */ 7 :
+      switch (lexeme[0]) {
+
+        case 'A' :
+          /* ARGLIST */
+          if (string.CompareOrdinal(lexeme, "ARGLIST") == 0) {
+            return Token.ARGLIST;
+          } /* end if */
+          break;
+
+        case 'P' :
+          /* POINTER */
+          if (string.CompareOrdinal(lexeme, "POINTER") == 0) {
+            return Token.POINTER;
+          } /* end if */
+          break;
+
+      } /* end switch */
+      break;
+
+    case /* length = 9 */ 9 :
+      switch (lexeme[0]) {
+
+        case 'P' :
+          /* PROCEDURE */
+          if (string.CompareOrdinal(lexeme, "PROCEDURE") == 0) {
+            return Token.PROCEDURE;
+          } /* end if */
+          break;
+
+        case 'Q' :
+          /* QUALIFIED */
+          if (string.CompareOrdinal(lexeme, "QUALIFIED") == 0) {
+            return Token.QUALIFIED;
+          } /* end if */
+          break;
+
+      } /* end switch */
+      break;
+
+    case /* length = 10 */ 10 :
+      /* DEFINITION */
+      if (string.CompareOrdinal(lexeme, "DEFINITION") == 0) {
+        return Token.DEFINITION;
+      } /* end if */
+      break;
+
+    case /* length = 14 */ 14 :
+      /* IMPLEMENTATION */
+      if (string.CompareOrdinal(lexeme, "IMPLEMENTATION") == 0) {
+        return Token.IMPLEMENTATION;
+      } /* end if */
+      break;
+
+  } /* end switch (length) */
+
+  return Token.Unknown;
+} /* end TokenForResword */
+
+
+/* ---------------------------------------------------------------------------
+ * method LexemeForResword(token)
+ * ---------------------------------------------------------------------------
+ * Returns a string with the lexeme for the reserved word represented by
+ * token.  Returns null if the token does not represent a reserved word.
+ * ------------------------------------------------------------------------ */
+
+public static string LexemeForResword (Token token) {
+  if (IsResword(token) == false) {
+    return null;
+  } /* end if */
+  return token.ToString();
+} /* end LexemeForResword */
```

Comment:
I created the array lexemeTable in TokenSet.cs to do the same job. I simply referenced lexemeTable in my fix, but because it's static I could easily move lexemeTable into Terminals.cs and then reference it as Terminals.lexemeTable in TokenSet if you would rather.